src/rollback_to_stable/rts_api.c

```diff
@@ -214,14 +222,22 @@ __rollback_to_stable_one(WT_SESSION_IMPL *session, const char *uri, bool *skipp)
       session, WT_VERB_RECOVERY_RTS(session), "starting rollback to stable on uri %s", uri);
 
     /* Read the stable timestamp once, when we first start up. */
-    WT_ACQUIRE_READ_WITH_BARRIER(rollback_timestamp, conn->txn_global.stable_timestamp);
+    WT_ACQUIRE_READ_WITH_BARRIER(stable_timestamp, conn->txn_global.stable_timestamp);
     WT_ACQUIRE_READ_WITH_BARRIER(pinned_timestamp, conn->txn_global.pinned_timestamp);
 
+    /* If the stable timestamp is not set, do not roll back based on it. */
+    rollback_timestamp = stable_timestamp;
+    if (rollback_timestamp == WT_TS_NONE) {
+        rollback_timestamp = WT_TS_MAX;
+        __wt_verbose_multi(session, WT_VERB_RECOVERY_RTS(session), WT_RTS_VERB_TAG_NO_STABLE "%s",
+          "the stable timestamp is not set; set the rollback timestamp to the maximum timestamp");
+    }
+
     F_SET(session, WT_SESSION_QUIET_CORRUPT_FILE);
     ret = __wti_rts_btree_walk_btree_apply(session, uri, config, rollback_timestamp);
```

Comment:
With this change of setting the `rollback_timestamp` to MAX value when there is no stable timestamp. We have a code in [here](https://github.com/wiredtiger/wiredtiger/blob/8a774da4ecf4a89a31487f1e21edbebbe8b025f6/src/rollback_to_stable/rts_btree_walk.c#L360) that also deals with the similar condition. You need to remove the second scenario from that as it will be a dead code with this change.

Reply:
That's a good catch! Fixed in [edf1709](https://github.com/wiredtiger/wiredtiger/pull/10580/commits/edf1709996c9292dc8d2ec2cba263432c0feb89c). I also added an assertion that the timestamp is set at this point.